osquery/config/parsers/options.cpp

```diff
@@ -37,38 +37,50 @@ class OptionsConfigParserPlugin : public ConfigParserPlugin {
     return {"options"};
   }
 
-  Status setUp() override;
-
   Status update(const std::string& source, const ParserConfig& config) override;
 };
 
-Status OptionsConfigParserPlugin::setUp() {
-  data_.put_child("options", pt::ptree());
-  return Status(0, "OK");
-}
-
 Status OptionsConfigParserPlugin::update(const std::string& source,
                                          const ParserConfig& config) {
   if (config.count("options") > 0) {
-    data_ = pt::ptree();
-    data_.put_child("options", config.at("options"));
+    if (!data_.doc().HasMember("options")) {
+      auto obj = data_.getObject();
+      data_.add("options", obj);
+    }
+
+    if (config.at("options").doc().IsObject()) {
+      auto obj = data_.getObject();
+      obj.CopyFrom(config.at("options").doc(), data_.doc().GetAllocator());
+      data_.mergeObject(data_.doc()["options"], obj);
+    }
   }
 
-  const auto& options = data_.get_child("options");
-  for (const auto& option : options) {
-    std::string value = options.get<std::string>(option.first, "");
-    if (value.empty()) {
+  const auto& options = data_.doc()["options"];
+  for (const auto& option : options.GetObject()) {
```

Comment:
It looks like that in this case you might be better off using the SAX parser.
I'm guessing that at this point the document is already parsed though.